File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string st) {
        stack<char>s;
        for(auto i : st){
            if(i == '(' or i == '[' or i == '{')
                s.push(i);
            else
            {
                if(s.empty() or (s.top() == '(' and i != ')') or (s.top() == '[' and i != ']') or (s.top() == '{' and i != '}')) return false;
                s.pop();
            }
        }
        return s.empty();
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = ['NULL']
        m = {')':'(' , ']':'[', '}':'{'}
        for i in s:
            if i in m:
                if stack.pop() != m[i]:
                    return False
            else:
                stack.append(i)
                
        return len(stack) == 1